Initializing a descriptor for Zenodo's DOI

castelao opened the pull request:
See issue #127 for discussion of these changes.

# Release checklist
- [ ] Authors updated in `cf-conventions.adoc`?
- [ ] Next version in `cf-conventions.adoc` up to date? Versioning inspired by [SemVer](https://semver.org).
- [ ] `history.adoc` up to date?
- [ ] Conformance document up-to-date?

# For maintainers
After the merge remember to delete the source branch.
Tags are set at the conclusion of the annual meeting; until then `master` always is a draft for the next version.


# Comments

@ethanrd and @davidhassell, since this came up in the last meeting, let's try it again? Just checking, as I recall the decision (from the governance panel long time ago) was to use Zenodo integrated with GitHub to make the recording and update process easier. If so, let's use a `.zenodo.json` descriptor to avoid manual editing in the future. Just a reminder, with that choice, we might not be able to edit all fields.

Note: This is just a skeleton just to guide it. It is missing fields, and records. For instance, who is going on the list of creators? I guess the natural choice would be the list of authors in the document.

EDIT 2023-11-8: Added issue number

ethanrd commented:
Hi Gui @castelao - Thanks for getting this started!
Here's the [comment](https://github.com/cf-convention/cf-conventions/issues/127#issuecomment-1085035774) on issue #127 that summarizes the CF Governance Panel's discussion/decision on DOIs.

castelao commented:
@ethanrd it's open for review. Note that it is missing several affiliations and ORCIDs.

castelao commented:
@larsbarring , I'm sorry. I let my branch fall behind, creating that confusion in your PR. To keep it simple, I rebased my branch and added manually your changes. Could you double check if I did it correctly? Thanks!

larsbarring commented on .zenodo.json:
```diff
@@ -0,0 +1,142 @@
+{
+  "title": "NetCDF Climate and Forecast (CF) Metadata Conventions",
+  "description": "Descriptive information: This document describes the CF conventions for climate and forecast metadata designed to promote the processing and sharing of files created with the netCDF Application Programmer Interface [NetCDF]. The conventions define metadata that provide a definitive description of what the data in each variable represents, and of the spatial and temporal properti ...
+  "license": "cc-zero",
```
As far as I understand CC0 is the chosen license, cf. [this issue comment](https://github.com/cf-convention/cf-convention.github.io/issues/182#issuecomment-1085036747). But there were some subsequent comments and I am not sure CC0 is the final decision. Hence I hand over this question to the Governance Panel (ping @ethanrd, @JonathanGregory, @bnlawrence, @erget, @taylor13, and @fannyadloff).

ethanrd replied:
I don't see any objections in the comments on using CC0. So I think we are good to go. I have summarized a few next steps on the license issue and will start working to add/specify the license.

sadielbartholomew commented on .zenodo.json:
```diff
@@ -0,0 +1,162 @@
+      "orcid": ""
+    },
+    {
+      "name": "Sadie Bartholomew",
+      "affiliation": "",
+      "orcid": ""
```
Hi all (@castelao, governance members), I had a skim read of this PR since it came up on my notifications and I thought whilst I was here I might as well update my own details. If you accept the suggestion here they can go into this PR. (If for some reason you don't want this yet then they can be added later, but these are the relevant values for the fields in my case.)

```suggestion
      "name": "Sadie L. Bartholomew",
      "affiliation": "National Centre for Atmospheric Science and University of Reading",
      "orcid": "0000-0002-6180-3603"
```

ethanrd replied:
Yes, I believe this is what we want to get for all the authors. Thanks!

ethanrd commented on .zenodo.json:
```diff
@@ -0,0 +1,162 @@
+{
+  "title": "NetCDF Climate and Forecast (CF) Metadata Conventions",
+  "description": "Descriptive information: This document describes the CF conventions for climate and forecast metadata designed to promote the processing and sharing of files created with the netCDF Application Programmer Interface [NetCDF]. The conventions define metadata that provide a definitive description of what the data in each variable represents, and of the spatial and temporal properti ...
```
We should drop the reference links - "[NetCDF]" and "[COARDS]" from the description text. Also, do we need the "Descriptive information:" at the beginning? It seems redundant.

davidhassell replied:
Could I suggest changing 

> The extensions include metadata that provides a precise definition of each variable via specification of a standard name, describes the vertical locations corresponding to dimensionless vertical coordinate values, and provides the spatial coordinates of non-rectilinear gridded data.

to 

"The extensions include metadata that provide a precise definition of each variable via specification of a standard name, describe the vertical locations corresponding to dimensionless vertical coordinate values, and provide the spatial coordinates of non-rectilinear gridded data, as well as data on unstructured meshes."

Also - just checking that the `\n` is intentional!

JonathanGregory replied:
Dear Gui

Thanks for working on this. I commented on this PR in the associated [issue](https://github.com/cf-convention/cf-conventions/issues/127#issuecomment-1921274631). Please could you look at that? In particular, I think that David's suggestion, while reasonable, should _not_ be made in this PR, because (if I understand correctly) the text you have used is the abstract of the conventions document.

Jonathan

castelao replied:
Shall I mark this 'conversation' as resolved?

ethanrd replied:
Works for me.

ethanrd reviewed (commented):
Looks good. Just the one comment on the description text.

JonathanGregory commented:
I suppose I should have put this here.

> Dear Gui
> 
> Thanks for working on this. I commented on this PR in the associated [issue](https://github.com/cf-convention/cf-conventions/issues/127#issuecomment-1921274631). Please could you look at that? In particular, I think that David's suggestion, while reasonable, should _not_ be made in this PR, because (if I understand correctly) the text you have used is the abstract of the conventions document.
> 
> Jonathan

JonathanGregory commented:
The reason for my commenting in the issue, rather than in the PR, is because that's generally preferred for CF discussions on substantive questions (as distinct from typos or other things that don't change the content or meaning significantly). That keeps the discussion all in one place, for easy reference and to make it more obvious to the whole community.

castelao commented:
@JonathanGregory , I'm sorry for my lack of attention on #127.

Just checking. Shall I revert [7a458440](https://github.com/cf-convention/cf-conventions/pull/443/commits/7a45844068835410d37399d6c3e1fc4b0a9a3c19)?

JonathanGregory commented:
> Just checking. Shall I revert [7a458440](https://github.com/cf-convention/cf-conventions/pull/443/commits/7a45844068835410d37399d6c3e1fc4b0a9a3c19)?

I'm not sure, I'm sorry to say! Am I right that you copied the text initially from the abstract of the conventions? If so, I think that was a good choice, and my opinion is that it would be best, at the moment, to use that text, after deleting the two "[...]" link texts, the "\n", and the "Descriptive information:" at the start. I think those were the editorial changes suggested by Ethan and David. David also suggested changing the content. As I wrote in #127, I feel it would be better to do that in a separate issue, and change it in the abstract as well, to keep them consistent. OK?

Many thanks again for your work on this.

castelao commented:
@ethanrd and @JonathanGregory , I think I covered all the requests. Please let me know if it is missing anything.

JonathanGregory commented:
Dear Gui @castelao 

Thanks for the updates. The text looks fine to me, except that Ethan suggested omitting "Descriptive information" at the start. Also, please see https://github.com/cf-convention/cf-conventions/issues/127 about the list of contributors.

Best wishes

Jonathan

ethanrd commented on .zenodo.json:
```diff
@@ -0,0 +1,208 @@
+{
+  "title": "NetCDF Climate and Forecast (CF) Metadata Conventions",
+  "description": "Descriptive information: This document describes the CF conventions for climate and forecast metadata designed to promote the processing and sharing of files created with the netCDF Application Programmer Interface. The conventions define metadata that provide a definitive description of what the data in each variable represents, and of the spatial and temporal properties of the ...
```
```suggestion
  "description": "This document describes the CF conventions for climate and forecast metadata designed to promote the processing and sharing of files created with the netCDF Application Programmer Interface. The conventions define metadata that provide a definitive description of what the data in each variable represents, and of the spatial and temporal properties of the data. This enables users of data from different sources to decide which quantities are comparable, and facilitates building applications with powerful extraction, regridding, and display capabilities. The CF conventions generalize and extend the COARDS conventions. The extensions include metadata that provides a precise definition of each variable via specification of a standard name, describes the vertical locations corresponding to dimensionless vertical coordinate values, and provides the spatial coordinates of non-rectilinear gridded data. Since climate and forecast data are often not simply representative of points in space/time, other extensions provide for the description of coordinate intervals, multidimensional cells and climatological time coordinates, and indicate how a data value is representative of an interval or cell. This standard also relaxes the COARDS constraints on dimension order and specifies methods for reducing the size of datasets.",
```

castelao replied:
@ethanrd , I'm sorry. I had edited that locally before, but seems like I forgot to push it. Could you confirm, please, if it is correct now? Thanks!

ethanrd replied:
Yes, looks good. Thanks Gui!

ethanrd commented on .zenodo.json:
```diff
@@ -0,0 +1,208 @@
+    {"name": "John Caron", "affiliation": "Unidata"},
+    {"name": "Antonio Cofi\u00f1o", "affiliation": "University of Cantabria"},
+    {"name": "Lorenzo Corgnati", "affiliation": "CNR-ISMAR"},
+    {"name": "Ethan Davis", "affiliation": "Unidata"},
```
```suggestion
    {"name": "Ethan Davis", "affiliation": "NSF Unidata", "orcid": "0000-0002-8530-2213"},
```

taylor13 replied:
Should we systematically include orcid i.d.?  If so, mine is 0000-0002-6491-2135.
Karl

ethanrd reviewed (commented):
Two minor changes: remove "Descriptive information" from start of abstract text; add my ORCID.

japamment commented on .zenodo.json:
```diff
@@ -0,0 +1,232 @@
+    },
+    {
+      "name": "Alison Pamment",
+      "affiliation": "CEDA"
```
"orcid": "0000-0001-5040-4626"

japamment replied:
Hi Gui @castelao

Thanks for working on this. I've added a comment with my ORCID - please could you include it in the pull request?

Best wishes,
Alison

castelao replied:
Could you double-check if it's correct, please?

erget commented:
@castelao here's my pitiful, unused ORCID: 0000-0002-1041-232X

mraspaud commented on .zenodo.json:
```diff
@@ -0,0 +1,234 @@
+    {"name": "Jean-Yves Peterschmitt", "affiliation": "LSCE-CEA-IPSL"},
+    {"name": "Alex Pletzer", "affiliation": "Tech-X Corporation"},
+    {"name": "Greg Rappa", "affiliation": "MIT"},
+    {"name": "Martin Raspaud", "affiliation": "SMHI"},
```
```suggestion
    {"name": "Martin Raspaud", "affiliation": "SMHI", "orcid": "0000-0002-3221-3778"},
```

castelao commented:
Hi everyone. Is this ready to move forward? It is still missing a few ORCIDs, but we can always add them later.

JonathanGregory commented:
Yes, I think it's ready. It would be great to close this ancient issue (#127)! Shall I merge this PR?
